pycbc/inference/models/hierarchical.py

```diff
@@ -919,14 +972,13 @@ def from_config(cls, cp, **kwargs):
         # it will not be listed in `variable_params` and `prior` sections
         primary_model = submodels[kwargs['primary_lbl'][0]]
         marginalized_params = primary_model.marginalize_vector_params.copy()
-        if 'logw_partial' in marginalized_params:
-            marginalized_params.pop('logw_partial')
-            marginalized_params = list(marginalized_params.keys())
-        else:
-            marginalized_params = []
-        # this may also include 'f_ref', 'f_lower', 'approximant',
-        # but doesn't matter
-        marginalized_params += list(primary_model.static_params.keys())
+        marginalized_params = list(marginalized_params.keys())
+        # add distance or phase if they are marginalized
+        if primary_model.distance_marginalization:
```

Comment:
Why do you need to hardcode this? This will be brittle and break easily to changes to marginalization for example, which we don't want. If you need a list of marginalized parameters, why not add this to the class in inference/tools so that it is kept up to date?

Reply:
OK, I will save it in the `tools` module and just use it here.